### PR TITLE
Fixing Issues

### PR DESCRIPTION
**Pull request by PrajvalRaval**
1) Updated Portuguese Model as suggested by Marcelo Schmidt.

2) Created separate intent for emoji message as it was rendering PostMessageIntent for non-english locale completely useless.

3) Fixed UnreadMessagesIntent which was not refactored properly.

4) Updated English Models for a separate Emoji Intent

### CODE REVIEW

**Review by acerbisgianluca — APPROVED** [review]

File: lambda/custom/helperFunctions.js [hunk 1]
```diff
@@ -163,20 +164,14 @@ const channelUnreadMessages = async (channelName, unreadCount, headers) =>
 					const msgs = [];
 
 					for (let i = 0; i <= unreadCount - 1; i++) {
-						msgs.push(
-							ri('GET_UNREAD_MESSAGES_FROM_CHANNEL.MESSAGE', {
-								name: res.messages[i].u.username,
-								message: res.messages[i].msg,
-							})
-						);
+						msgs.push(`${res.messages[i].u.username} says, ${res.messages[i].msg} <break time="0.7s"/> `);
```
Comment (TheDreamSaver) [inline lambda/custom/helperFunctions.js:167]: "says" word will not be localized here. A minor issue, for now, but should definitely add it to the list.

**Review by TheDreamSaver — COMMENTED** [review]
We can go with a different intent for emojis but not sure if it can improve recognition as the audio model is still similar.
What was the error you were getting in case of a single intent?

File: lambda/custom/helperFunctions.js (hunk 1 above)
Comment (PrajvalRaval) [reply lambda/custom/helperFunctions.js:167]: > "says" word will not be localized here. A minor issue, for now, but should definitely add it to the list.

Sure

**Comment by PrajvalRaval** [comment]
> We can go with a different intent for emojis but not sure if it can improve recognition as the audio model is still similar.
> What was the error you were getting in case of a single intent?

In single intent, when you try to invoke PostMessageIntent it simply returns an error. For example in Portuguese when I tried to invoke by "Alexa, Mandar ola Para O general" the message was not being send to channel and the only response I receive was "Desculpe, não consegui enviar sua mensagem agora." which is an error response. You can try it on your own using our master branch code. Besides two intents will not be a major difference for UX, so I thinks a separate intent is still a better choice.

**Comment by TheDreamSaver** [comment]
I was able to send emoji using the same PostMessageIntent for the Portuguese model.
The reason you're getting the error because the master code doesn't have the emoji slot and utterances supporting that in the Portuguese as well as other non-english locale models.
You can add these utterances in the master PostMessageIntent,

`mandar {messagepost} emoji {emoji} para o {messagechannel}`

`mandar {messagepost} emoji {emoji} {emojiTwo} para o {messagechannel}`

`mandar {messagepost} emoji {emoji} {emojiTwo} {emojiThree} para o {messagechannel}`

`mandar {messagepost} para o {messagechannel}`

and also add the emoji slotType and then test again.
You won't get the error anymore.

**Comment by PrajvalRaval** [comment]
> I was able to send emoji using the same PostMessageIntent for the Portuguese model.
> The reason you're getting the error because the master code doesn't have the emoji slot and utterances supporting that in the Portuguese as well as other non-english locale models.
> You can add these utterances in the master PostMessageIntent,
> 
> `mandar {messagepost} emoji {emoji} para o {messagechannel}`
> 
> `mandar {messagepost} emoji {emoji} {emojiTwo} para o {messagechannel}`
> 
> `mandar {messagepost} emoji {emoji} {emojiTwo} {emojiThree} para o {messagechannel}`
> 
> `mandar {messagepost} para o {messagechannel}`
> 
> and also add the emoji slotType and then test again.
> You won't get the error anymore.

Your are correct that adding emoji slot will work, but then we will have to add emoji slot to every locale which we are waiting for native speakers to review and if we do not add then beginner contributors will face an error in PostMessageIntent, while separate intent will prevent this error from everyone and contributors can decide to work on creating emoji intent as per their choice.

**Review by TheDreamSaver — APPROVED** [review]

**Pull request merged**